### PR TITLE
feat(unmapping): adds inter-convertible optional and nullable values for responses

### LINKED ISSUE #169: Relax unmapping of optional values to nullable types and vice versa
We need the following unmapping cases relaxed:

#### Optional to Nullable

Optional values should be unmapped into nullable types i.e. `undefined` should be assignable to `string | nullable` during response validation

#### Nullable to Optional

Nullable values should be unmapped into optional types i.e. `null` should be assignable to `string | undefined` during response validation

### PR DESCRIPTION
**Pull request by asadali214**
## What

See the attached issue for details

Closes #169

### CODE REVIEW

File: packages/schema/src/utils.ts [hunk 1]
```diff
@@ -151,3 +151,7 @@ export function omitKeysFromObject(
 export function objectKeyEncode(key: string): string {
   return key.indexOf(' ') !== -1 ? literalToString(key) : key;
 }
+
+export function isNullOrMissing(value: unknown): value is null | undefined {
+  return value === null || value === undefined;
```
Comment (MaryamAdnan3) [inline packages/schema/src/utils.ts:156]: its generally considered a good practice to use the `typeof `operator check the type undefined i.e `return typeof value === 'undefined' || value === null;`

**Review by MaryamAdnan3 — CHANGES_REQUESTED** [review]
Please see the comments

File: packages/schema/src/utils.ts (hunk 1 above)
Comment (asadali214) [reply packages/schema/src/utils.ts]: Added suggestion

File: packages/schema/test/types/optional.test.ts [hunk 2]
```diff
@@ -17,6 +17,13 @@ describe('Optional', () => {
       expect((output as any).result).toBeUndefined();
     });
 
+    it('should accept null', () => {
```
[the remark this answers is not in the document]
Comment (asadali214) [reply packages/schema/test/types/optional.test.ts]: The schema function validateAndMap() is testing both validateBeforeMap and map functionality

[the remark this answers is not in the document]
Comment (asadali214) [reply packages/schema/test/types/optional.test.ts]: Done

**Review by MaryamAdnan3 — DISMISSED** [review]
LGTM

**Pull request merged**